presto-hive/src/main/java/io/prestosql/plugin/hive/HiveBucketValidationRecordCursor.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.VerifyException;
+import io.airlift.slice.Slice;
+import io.prestosql.plugin.hive.util.ForwardingRecordCursor;
+import io.prestosql.plugin.hive.util.HiveBucketing.BucketingVersion;
+import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.block.Block;
+import io.prestosql.spi.connector.RecordCursor;
+import io.prestosql.spi.type.Type;
+import io.prestosql.spi.type.TypeManager;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfo;
+
+import java.util.List;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.prestosql.plugin.hive.HiveErrorCode.HIVE_INVALID_BUCKET_FILES;
+import static io.prestosql.plugin.hive.util.HiveBucketing.getHiveBucket;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class HiveBucketValidationRecordCursor
+        extends ForwardingRecordCursor
+{
+    private final RecordCursor delegate;
+    private final Path path;
+    private final int[] bucketColumnIndices;
+    private final List<Class<?>> javaTypeList;
+    private final List<TypeInfo> typeInfoList;
+    private final BucketingVersion bucketingVersion;
+    private final int bucketCount;
+    private final int expectedBucket;
+
+    private final Object[] scratch;
```

Comment:
maybe add a comment what this is & what is for, as the name is not fully self-describing

Reply:
This is copied from `HiveBucketAdapterRecordCursor`. If you look at the only method in this class (other than the constructor), the usage seems obvious.